src/cdk-experimental/menu/menu-item.ts

```diff
@@ -208,17 +172,41 @@ export class CdkMenuItem implements FocusableOption, FocusableElement, Toggler,
     }
   }
 
-  /**
-   * Subscribe to the mouseenter events and close any sibling menu items if this element is moused
-   * into.
-   */
-  private _setupMouseEnter() {
+  private _setupEventListeners() {
+    const element = this._elementRef.nativeElement;
+
```

Comment:
As mentioned in other PRs, all of the extra code due to this boilerplate isn't worth the reduction in change detections. These detections only happen for a small subtree of the DOM and only when the user is actually interacting with a component so it's unlikely to cause any issues.

Reply:
Well, I see the same pattern is being used in many places of the code and its also being added in newer PRs, I mean adding event listeners outside of the zone.

I'm a bit unaware what means _small subtree_, if the consumer doesn't use OnPush on his components then the change detection will happen on the whole tree. This, of course, won't happen on material components, since they're mostly all marked with OnPush.

Reply:
We do this for events that can happen frequently (e.g. `scroll` or `mousemove`), but for something like `focus` and `blur` it's unlikely to make much of a difference since they can only happen on one element at a time, only while a menu is open.